src/Bicep.Core/Analyzers/Linter/Common/LinterExpressionHelper.cs

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Navigation;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Metadata;
+using Bicep.Core.Syntax;
+using Bicep.Core.Syntax.Visitors;
+using Bicep.Core.TypeSystem.Az;
+using Microsoft.WindowsAzure.ResourceStack.Common.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Bicep.Core.Analyzers.Linter.Common
+{
+    internal static class LinterExpressionHelper
+    {
+        /// <summary>
+        /// Tries to retrieve a string literal from the expression. Will evaluate variables and parameter default values
+        /// </summary>
+        // TODO: Refactor more rules to use this
+        public static (string stringValue, StringSyntax stringSyntax, string? pathToValueIfNonTrivial)?
```

Comment:
nit: formatting is off here

Reply:
Otherwise the function name is far to the right, I prefer this